src/compiler/llvm/mir.rs

```diff
@@ -0,0 +1,215 @@
+//! Transforms the MIR representation into LLVM
+
+use std::collections::{hash_map::Entry, HashMap};
+
+use inkwell::{builder::Builder, context::Context, module::Module, values::*};
+
+use crate::{
+    compiler::{
+        mir::{ir::*, Transformer},
+        Span,
+    },
+    StringId, StringTable,
+};
+
+struct LlvmTransformer<'ctx> {
+    context: &'ctx Context,
+    module: Module<'ctx>,
+    builder: Builder<'ctx>,
```

Comment:
Module and Builder should be passed in to the Transformer.  It does not manage the LLVM objects it only uses them to transform MIR to LLVM.